File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void helper(int &sum, int ans, TreeNode* root){
        if(!root) return;
        ans = ans* 10 + root->val;
        if(root->left == NULL && root->right == NULL)
            sum += ans;
        helper(sum, ans, root->left);
        helper(sum, ans, root->right);
    }
    int sumNumbers(TreeNode* root) {
        int sum = 0;
        int ans = 0;
        helper(sum, ans, root);
        return sum;
        
    }
};

File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    void helper(int[] sum, int ans, TreeNode root)
    {
        if(root == null)
            return;
        ans = ans * 10 + root.val;
        if(root.left == null && root.right == null)
        {
            sum[0] += ans;
            return;
        }
        helper(sum, ans, root.left);
        helper(sum, ans, root.right);
    }
    public int sumNumbers(TreeNode root) {
        int sum[] = new int[1];
        int ans = 0;
        helper(sum, ans, root);
        return sum[0];
    }
}